flink-runtime/src/test/java/org/apache/flink/runtime/io/disk/FileChannelManagerImplTest.java

```diff
@@ -0,0 +1,192 @@
+		assumeTrue(OperatingSystem.isLinux()
+				|| OperatingSystem.isFreeBSD()
+				|| OperatingSystem.isSolaris()
+				|| OperatingSystem.isMac());
+
+		File fileChannelDir = temporaryFolder.newFolder();
+		File signalDir = temporaryFolder.newFolder();
+
+		FileChannelManagerTestProcess fileChannelManagerTestProcess = null;
+
+		String javaCommand = getJavaCommandPath();
+		if (javaCommand == null) {
+			fail("Could not find java executable.");
+		}
+
+		try {
+			fileChannelManagerTestProcess = new FileChannelManagerTestProcess(
+					callerHasHook,
+					fileChannelDir.getAbsolutePath(),
+					FilenameUtils.concat(signalDir.getAbsolutePath(), COULD_KILL_SIGNAL_FILE));
+			fileChannelManagerTestProcess.startProcess();
+
+			// Waits till netty shuffle environment has created the tmp directories.
+			Deadline deadline = Deadline.now().plus(TEST_TIMEOUT);
+
+			while (!fileOrDirExists(signalDir, COULD_KILL_SIGNAL_FILE) && deadline.hasTimeLeft()) {
+				Thread.sleep(100);
+			}
+
+			if (!fileOrDirExists(signalDir, COULD_KILL_SIGNAL_FILE) ||
+					!fileOrDirExists(fileChannelDir, DIR_NAME_PREFIX)) {
+				fail("The file channel manager test process does not create target directories in time, " +
+						"its output is: \n" + fileChannelManagerTestProcess.getProcessOutput());
+			}
+
+			Process kill = Runtime.getRuntime().exec("kill " + fileChannelManagerTestProcess.getProcessId());
+			kill.waitFor();
+			assertEquals("Failed to send SIG_TERM to process", 0, kill.exitValue());
+
+			deadline = Deadline.now().plus(TEST_TIMEOUT);
+			while (fileChannelManagerTestProcess.isAlive() && deadline.hasTimeLeft()) {
+				Thread.sleep(100);
+			}
+
+			if (fileChannelManagerTestProcess.isAlive()) {
+				fail("The file channel manager test process does not terminate in time, its output is: \n" + fileChannelManagerTestProcess.getProcessOutput());
+			}
+
+			// Checks if the directories are cleared.
+			assertThat("The file channel manager test process does not remove the tmp shuffle directories after termination, " +
+							"its output is \n" + fileChannelManagerTestProcess.getProcessOutput(),
+					fileOrDirExists(fileChannelDir, DIR_NAME_PREFIX),
+					is(false));
+		} finally {
+			if (fileChannelManagerTestProcess != null) {
+				fileChannelManagerTestProcess.destroy();
+			}
+		}
+	}
+
+	private boolean fileOrDirExists(File rootTmpDir, String namePattern) {
+		File[] candidates = rootTmpDir.listFiles((dir, name) -> name.contains(namePattern));
+		return candidates != null && candidates.length > 0;
+	}
+
+	/**
+	 * The {@link FileChannelManagerCleanupRunner} instance running in a separate JVM process.
+	 */
+	private static class FileChannelManagerTestProcess extends TestJvmProcess {
+		private final boolean callerHasHook;
+		private final String tmpDirectories;
+		private final String couldKillSignalFilePath;
+
+		public FileChannelManagerTestProcess(boolean callerHasHook, String tmpDirectories, String couldKillSignalFilePath) throws Exception {
```

Comment:
can be package private and remove the `throws Exception`

Reply:
Agree with that, and the code has been modified accordingly.